pkg/sessiontxn/staleread/util.go

```diff
@@ -49,6 +51,11 @@ func CalculateAsOfTsExpr(ctx context.Context, sctx sessionctx.Context, tsExpr as
 		return 0, errAsOf.FastGenWithCause("as of timestamp cannot be NULL")
 	}
 
+	// if tsVal is TSO already, return it directly.
+	if tso, err := strconv.ParseUint(tsVal.GetString(), 10, 64); err == nil {
```

Comment:
I think it's inaccurate to check if it's a valid ts in this way, better to check if there's a related utility function.